op-e2e/config/init.go

```diff
@@ -32,34 +27,94 @@ const (
 	LegacyLevelTrace
 )
 
+type AllocType string
+
+const (
+	AllocTypeStandard AllocType = "standard"
+	AllocTypeAltDA    AllocType = "alt-da"
+	AllocTypeL2OO     AllocType = "l2oo"
+	AllocTypeMTCannon AllocType = "mt-cannon"
+
+	DefaultAllocType = AllocTypeStandard
+)
+
+func (a AllocType) Check() error {
+	switch a {
+	case AllocTypeStandard, AllocTypeAltDA, AllocTypeL2OO, AllocTypeMTCannon:
+		return nil
+	default:
+		return fmt.Errorf("unknown alloc type: %q", a)
+	}
+}
+
+func (a AllocType) SupportsProofs() bool {
+	switch a {
+	case AllocTypeStandard, AllocTypeMTCannon:
+		return true
+	default:
+		return false
+	}
+}
```

Comment:
We actually use proofs for everything except the L2OO variant. Things like alt-da which don't have support in op-program can't handle having games actually get to the point of running cannon but work fine in e2e tests where that doesn't happen and can use the permissioned game still.  I'd also suggest calling this `UsesProofs` since it's possible to have configs that support proofs but aren't actually using them for withdrawals.